File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def preorder(node):
            if not node:
                return None
            
            node.left, node.right = node.right, node.left
            preorder(node.left)
            preorder(node.right)
            return root
        return preorder(root)
            

            

File: 0226-invert-binary-tree/0226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {

        return preorder(root);
    }
    public TreeNode preorder(TreeNode node) {
            if(node == null) {
                return null;
            }

            TreeNode temp = node.left;
            node.left = node.right;
            node.right = temp;
            preorder(node.left);
            preorder(node.right);

            return node;
        }
}